crates/core/src/surfnet/locker.rs

```diff
@@ -1124,6 +1127,22 @@ impl SurfnetSvmLocker {
                 meta,
             )));
 
+            // Record the failed tx locally
+            self.with_svm_writer(|svm_writer| {
+                let twsm = TransactionWithStatusMeta::from_failure(
+                    simulated_slot,
+                    transaction,
+                    &failed_transaction_metadata,
+                    accounts_before,
+                    loaded_addresses.clone().unwrap_or_default(),
+                    0,
+                );
+                svm_writer.transactions.insert(
```

Comment:
A transaction that fails during simulation is _not_ processed, and shouldn't be inserted. That's the point of the preflight check - simulate the transaction before executing so I don't make state changes/spend any funds on a transaction that will fail.

All of the changes to this `handle_simulation_failure` can be reverted